contributors/design-proposals/selector-generation-from-kubectl.md

```diff
@@ -0,0 +1,174 @@
+Selector generation from kubectl
+==================
+
+# Goals
+
+Make `kubectl apply` work well with selector.
```

Comment:
Agreed with @soltysh. Also I believe selector generation is going to be useful for more than just making `kubectl apply` work. For example, in Deployments we already need to generate a label for making the underlying ReplicaSets non-overlapping.